packages/js/product-editor/src/wp-hooks/hide-inventory-advanced-collapsible.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import type { BlockInstance } from '@wordpress/blocks';
+import { createElement } from '@wordpress/element';
+import { addFilter } from '@wordpress/hooks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { select } from '@wordpress/data';
+import { evaluate } from '@woocommerce/expression-evaluation';
+
+/**
+ * Internal dependencies
+ */
+import { useEvaluationContext } from '../utils';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const areAllBlocksInvisible = ( blocks: BlockInstance[], context: any ) => {
+	return blocks.every( ( block ) => {
+		if (
+			! block.attributes?._templateBlockHideConditions ||
+			! Array.isArray( block.attributes?._templateBlockHideConditions )
+		) {
+			return false;
+		}
+		return block.attributes._templateBlockHideConditions.some(
+			( condition ) => evaluate( condition.expression, context )
+		);
+	} );
+};
+
+const maybeHideInventoryAdvancedCollapsible = createHigherOrderComponent<
+	Record< string, unknown >
+>( ( BlockEdit ) => {
+	return ( props ) => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const evalContext = useEvaluationContext( props.context as any );
+
+		if (
+			( props?.attributes as Record< string, string > )
+				?._templateBlockId !== 'product-inventory-advanced'
+		) {
+			return <BlockEdit { ...props } />;
+		}
+
+		// get the inventory section block instance
+		const advancedCollapsibleBlock = select( 'core/block-editor' ).getBlock(
```

Comment:
Using this version of `select` doesn't work well with interactivity when updating the form, we should really be using the `useSelect` hook and run this logic inside the callback, something like this ( btw I quickly wrote this, so maybe the logic can be adjusted a bit ):
```
const { hasInnerBlocks, allBlocksInvisible: blocksInvisible } =
			useSelect( ( select ) => {
				const advancedCollapsibleBlock = select(
					'core/block-editor'
				).getBlock( props?.clientId as string );

				let allBlocksInvisible = false;
				if ( advancedCollapsibleBlock?.innerBlocks?.length ) {
					const advancedSectionBlock =
						advancedCollapsibleBlock?.innerBlocks[ 0 ];
					allBlocksInvisible = areAllBlocksInvisible(
						advancedSectionBlock?.innerBlocks,
						evalContext.getEvaluationContext( select )
					);
				}

				return {
					hasInnerBlocks:
						! advancedCollapsibleBlock?.innerBlocks?.length,
					allBlocksInvisible,
				};
			} );
```

At the moment this causes an issue when enabling track stock, which adds extra items in the advanced dropdown, which in theory should make it show up. Using the `useSelect` helps solve this.
https://github.com/woocommerce/woocommerce/assets/2240960/4f0ab24f-97e5-418c-a406-daeb29bd6ec3